c/jesrequestsjson.c

```diff
@@ -0,0 +1,490 @@
+                                             errorBuffer, sizeof(errorBuffer));
+    if (json) {
+      if (jsonIsObject(json)) {
+        JsonObject *jsonObject = jsonAsObject(json);
+        char *fileValue = jsonObjectGetString(jsonObject,"file");
+        zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "fileValue...(%s)\n", fileValue);
+        if (fileValue) {
+          snprintf(fileName, sizeof(fileName), "//'%s'", fileValue);
+          zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "filename...(%s)\n", fileName);
+          respondWithSubmitDataset(response, fileName, TRUE);
+        }
+        else {
+          char *jclValue = jsonObjectGetString(jsonObject,"jcl");
+          if (jclValue) {
+            if (jclValue[0] != '/' | jclValue[1] != '/') {
+              respondWithMessage(response, HTTP_STATUS_INTERNAL_SERVER_ERROR,
+                  "BAD_REQUEST failed with RC = %d, Submit input data does not start with a slash", 8);
+            }
+            else {
+              responseWithSubmitJCLContents(response, jclValue);
+            }
+          }
+          else {
+            zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "jclValue...(%s)\n", jclValue);
+            zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "No jcl Value...(%s)\n", jclValue);
+            respondWithError(response, HTTP_STATUS_BAD_REQUEST,"POST body No jcl value");   
+          }   
+
+        }
+        /* updateDatasetWithJSON(response, jsonObject, absolutePath, etag, force); */
+      } else {
+        zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "*** INTERNAL ERROR *** message is JSON, but not an object\n");
+      }
+    }
+    else {
+      zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "UPDATE DATASET: body was not JSON!\n");
+      respondWithError(response, HTTP_STATUS_BAD_REQUEST,"POST body could not be parsed as JSON format");      
+    }
+    SLHFree(slh);
+  }
+  else {
+    respondWithError(response,HTTP_STATUS_INTERNAL_SERVER_ERROR,convertedBody);    
+  }
+  safeFree(convertedBody,conversionBufferLength);
+
+#endif /* __ZOWE_OS_ZOS */
+}
+
+
+static void respondWithSubmitDatasetInternal(HttpResponse* response,
+                                       const char *datasetPath,
+                                       const DatasetName *dsn,
+                                       const DDName *ddName,
+                                       int jsonMode) {
+#ifdef __ZOWE_OS_ZOS
+  HttpRequest *request = response->request;
+
+  char ddPath[16];
+  snprintf(ddPath, sizeof(ddPath), "DD:%8.8s", ddName->value);
+
+  int lrecl = getLreclOrRespondError(response, dsn, ddPath);
+  if (!lrecl) {
+    return;
+  }
+
+  if (lrecl){
+    zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "Streaming data to INTRDR for %s\n", datasetPath);
+    
+    /* jsonStart(jPrinter); */
+    int status = streamDatasetToINTRDR(response, ddPath, lrecl, TRUE);
+    /* jsonEnd(jPrinter); */
+  }
+
```

Comment:
This is a bit confusing, who will eventually respond to the caller? All the `respondXXXX` calls should be clearly seen, otherwise it's hard to see that all the branches result in a response (be it a success or failure).

Reply:
Updated the code based on the comment